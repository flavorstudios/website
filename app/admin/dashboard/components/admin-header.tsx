"use client"

import { Button } from "@/components/ui/button"
import { Input } from "@/components/ui/input"
import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar"
import { Bell, Search, Menu, ExternalLink, LogOut } from "lucide-react"

interface AdminHeaderProps {
  onLogout: () => void
  sidebarOpen: boolean
  setSidebarOpen: (open: boolean) => void
}

export function AdminHeader({ onLogout, sidebarOpen, setSidebarOpen }: AdminHeaderProps) {
  return (
    <header className="bg-white border-b border-gray-200 px-6 py-4">
      <div className="flex items-center justify-between">
        <div className="flex items-center gap-4">
          <Button variant="ghost" size="sm" onClick={() => setSidebarOpen(!sidebarOpen)} className="lg:hidden">
            <Menu className="h-5 w-5" />
          </Button>

          <div className="hidden md:flex items-center gap-4">
            <h1 className="text-xl font-bold bg-gradient-to-r from-purple-600 to-blue-600 bg-clip-text text-transparent">
              Flavor Studios Admin
            </h1>
            <div className="relative">
              <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 h-4 w-4" />
              <Input placeholder="Search..." className="pl-10 w-64" />
            </div>
          </div>
        </div>

        <div className="flex items-center gap-4">
          <Button
            variant="outline"
            size="sm"
            onClick={() => window.open("https://flavorstudios.in", "_blank")}
            className="hidden md:flex items-center gap-2"
          >
            <ExternalLink className="h-4 w-4" />
            View Site
          </Button>

          <Button variant="ghost" size="sm" className="relative">
            <Bell className="h-5 w-5" />
            <span className="absolute -top-1 -right-1 bg-red-500 text-white text-xs rounded-full h-5 w-5 flex items-center justify-center">
              3
            </span>
          </Button>

          <div className="flex items-center gap-3">
            <Avatar className="h-8 w-8">
              <AvatarImage src="/admin-avatar.jpg" />
              <AvatarFallback className="bg-gradient-to-br from-purple-600 to-blue-600 text-white text-sm">
                FS
              </AvatarFallback>
            </Avatar>
            <div className="hidden md:block">
              <p className="text-sm font-medium">Admin</p>
              <p className="text-xs text-gray-500">Flavor Studios</p>
            </div>
          </div>

          <Button variant="ghost" size="sm" onClick={onLogout} className="text-red-600 hover:text-red-700">
            <LogOut className="h-4 w-4" />
          </Button>
        </div>
      </div>
    </header>
  )
}
