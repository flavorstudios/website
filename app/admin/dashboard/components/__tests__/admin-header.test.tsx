import { render } from "@testing-library/react"
import { AdminHeader } from "../admin-header"

jest.mock("next/navigation", () => ({
  usePathname: () => "/admin/dashboard",
  useRouter: () => ({ push: jest.fn(), prefetch: jest.fn() }),
}))

// Make next/dynamic return a no-op component for tests
jest.mock("next/dynamic", () => () => () => null)

describe("AdminHeader", () => {
  beforeEach(() => {
    // stub fetch used for avatar + notifications loading
    global.fetch = jest.fn((url: RequestInfo) => {
      const s = typeof url === "string" ? url : ""
      if (s.includes("notifications")) {
        return Promise.resolve({
          ok: true,
          json: async () => ({ notifications: [] }),
        }) as unknown as Response
      }
      return Promise.resolve({
        ok: true,
        json: async () => ({ settings: { profile: { avatar: "" } } }),
      }) as unknown as Response
    }) as unknown as typeof fetch
  })

  it("matches snapshot", () => {
    const { container } = render(
      <AdminHeader
        onLogout={() => {}}
        sidebarOpen={false}
        setSidebarOpen={() => {}}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
