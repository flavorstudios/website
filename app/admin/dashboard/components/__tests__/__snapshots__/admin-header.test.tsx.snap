// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AdminHeader matches snapshot 1`] = `
<div>
  <a
    class="sr-only focus:not-sr-only focus:absolute focus:top-2 focus:left-2 focus:z-50 focus:rounded focus:bg-primary focus:text-primary-foreground focus:p-2 focus:outline-none focus:ring"
    href="#main"
  >
    Skip to main content
  </a>
  <header
    class="sticky top-0 z-40 h-14 px-4 flex items-center gap-2 border-b pt-[env(safe-area-inset-top)] backdrop-blur supports-[backdrop-filter]:bg-white/70 dark:supports-[backdrop-filter]:bg-zinc-900/70"
    role="banner"
  >
    <div
      class="flex flex-wrap items-center justify-between gap-2 sm:gap-4 w-full"
    >
      <button
        class="sr-only"
        type="button"
      >
        Open command palette
      </button>
      <div
        class="flex items-center flex-1 min-w-0"
      >
        <button
          aria-controls="admin-sidebar"
          aria-expanded="false"
          aria-label="Open sidebar"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-9 lg:hidden min-h-11 px-4 rounded-lg focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-menu h-5 w-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="20"
              y1="12"
              y2="12"
            />
            <line
              x1="4"
              x2="20"
              y1="6"
              y2="6"
            />
            <line
              x1="4"
              x2="20"
              y1="18"
              y2="18"
            />
          </svg>
        </button>
        <div
          aria-hidden="true"
          class="mx-4 h-10 w-px bg-gray-200 dark:bg-gray-700 hidden sm:block"
        />
        <h1
          class="text-xl font-bold bg-gradient-to-r from-purple-600 to-blue-600 bg-clip-text text-transparent whitespace-nowrap forced-colors:text-current"
        >
          Flavor Studios Admin
        </h1>
      </div>
      <div
        class="flex flex-wrap items-center gap-2 sm:gap-4"
      >
        <button
          aria-label="Search"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent h-10 w-10 text-sm font-medium transition-colors hover:text-blue-600"
        >
          <svg
            class="lucide lucide-search h-5 w-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <path
              d="m21 21-4.3-4.3"
            />
          </svg>
        </button>
        <button
          aria-label="Open live site in a new tab"
          class="justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 hidden md:flex items-center gap-2 min-h-11 px-4 rounded-lg focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-external-link h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 3h6v6"
            />
            <path
              d="M10 14 21 3"
            />
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
          </svg>
          View Site
        </button>
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          aria-label="Quick actions"
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 gap-2"
          data-state="closed"
          id="radix-«r0»"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-plus h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="M12 5v14"
            />
          </svg>
          <span
            class="hidden sm:inline"
          >
            New
          </span>
        </button>
        <div
          class="relative"
        >
          <button
            aria-expanded="false"
            aria-haspopup="dialog"
            aria-label="Notifications"
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground px-3 relative h-11 w-11 rounded-full focus:outline-none focus-visible:ring-2 focus-visible:ring-ring"
            type="button"
          >
            <svg
              class="lucide lucide-bell h-5 w-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 8a6 6 0 0 1 12 0c0 7 3 9 3 9H3s3-2 3-9"
              />
              <path
                d="M10.3 21a1.94 1.94 0 0 0 3.4 0"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex items-center gap-3"
        >
          <span
            class="relative flex shrink-0 overflow-hidden rounded-full h-8 w-8"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-muted bg-gradient-to-br from-purple-600 to-blue-600 text-white text-sm"
            >
              FS
            </span>
          </span>
          <div
            class="hidden md:block"
          >
            <p
              class="text-sm font-medium text-foreground"
            >
              Admin
            </p>
            <p
              class="text-xs text-muted-foreground"
            >
              Flavor Studios
            </p>
          </div>
        </div>
        <button
          aria-label="Toggle theme"
          aria-pressed="false"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 focus:outline-none focus:ring"
          title="Toggle theme (currently light)"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-moon h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
          <span
            class="sr-only"
          >
            Switch to 
            dark
             theme
          </span>
        </button>
        <button
          aria-label="Logout"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent h-9 text-red-600 hover:text-red-700 min-h-11 px-4 rounded-lg focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-log-out h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
</div>
`;
